tools/go-generator/cmd/generator/main.go

```diff
@@ -0,0 +1,362 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/format"
+	"io"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"github.com/tallstoat/pbparser"
+)
+
+var typeMapForRepeatedFields = map[string]string{
+	"google.protobuf.StringValue": "wrappers.String",
+}
+
+func getGoPackageName(pf pbparser.ProtoFile) string {
+	for _, opt := range pf.Options {
+		if opt.Name == "go_package" {
+			return opt.Value
+		}
+	}
+	return pf.PackageName
+}
+
+var zeroValues = map[string]string{
+	"string": `""`,
+	"bool":   `false`,
+}
+
+func toPublicFieldName(name string) string {
+	return strings.Title(name)
+}
+
+func isEnum(pf pbparser.ProtoFile, name string) bool {
+	for _, e := range pf.Enums {
+		if e.Name == name {
+			return true
+		}
+	}
+
+	return false
+}
+
+func shouldSkipOtherLanguageAgent(typeName string) bool {
+	return strings.HasSuffix(typeName, "Agent") && !strings.HasPrefix(typeName, "Go")
+}
+
+func main() {
+	var outDir = flag.String("out", ".", "Out directory for the generated code.")
+	var optModule = flag.String("opt-module", ".", "Module for the generated code.")
+	var envPrefix = flag.String("env-prefix", "HT_", "Prefix for env var loading")
```

Comment:
Should we keep this around or do we want to change to TA? Backwards compatibility could force us to keep it around.